src/Microsoft.ML.Core/Utilities/Utils.cs

```diff
@@ -247,8 +247,18 @@ public static bool TryFindIndexSorted(this int[] input, int min, int lim, int va
         /// </summary>
         public static int FindIndexSorted(this int[] input, int min, int lim, int value)
         {
-            Contracts.AssertValueOrNull(input);
-            Contracts.Assert(0 <= min & min <= lim & lim <= Utils.Size(input));
+            return FindIndexSorted(input.AsSpan(), min, lim, value);
```

Comment:
>input.AsSpan() [](start = 35, length = 14)

How heavy is AsSpan operation?
I would rather had two copies of same code for Array and ReadOnlySpan than perf degradation.

Reply:
> How heavy is AsSpan operation?

It is super lightweight. There is actually in implicit conversion between array and span. It is only necessary here to tell the compiler to call the Span overload instead of the array overload.